src/AzureRMAssets.ts

```diff
@@ -26,46 +27,29 @@ export class AzureRMAssets {
         return "azuresdk-3.0.0";
     }
 
-    public static getFunctionMetadata(): Promise<FunctionMetadata[]> {
+    public static getFunctionMetadata(): Promise<FunctionsMetadata> {
         if (AzureRMAssets._functionMetadata === undefined) {
             AzureRMAssets._functionMetadata = AzureRMAssets.getFunctionMetadataUri()
                 .then(AzureRMAssets.readFile)
                 .then(FunctionMetadata.fromString);
         }
-        return AzureRMAssets._functionMetadata;
-    }
-
-    public static getFunctionMetadataWithName(functionName: string): Promise<FunctionMetadata> {
-        return AzureRMAssets.getFunctionMetadata()
-            .then((functionMetadataArray: FunctionMetadata[]) => {
-                let result: FunctionMetadata = null;
 
-                const lowerCasedFunctionName: string = functionName.toLowerCase();
-                for (const functionMetadata of functionMetadataArray) {
-                    if (functionMetadata.name && functionMetadata.name.toLowerCase() === lowerCasedFunctionName) {
-                        result = functionMetadata;
-                        break;
-                    }
-                }
+        if (!AzureRMAssets._functionsMetadata) {
+            AzureRMAssets._functionsMetadata = AzureRMAssets._functionMetadata
+                .then((array: FunctionMetadata[]) => {
+                    return new FunctionsMetadata(array);
+                });
+        }
```

Comment:
I see you use async-await in another file. Considering consistency, what's the reasoning behind using then(s) here?

Reply:
Not my code.  Don't want to change everything to async just because I'm changing part of the function.